controlplane/kubeadm/api/.import-restrictions

```diff
@@ -0,0 +1,5 @@
+rules:
```

Comment:
Let's also add a import-restrictions file to the v1beta1 package to block controller-runtime imports entirely there (like we did for the bootstrap provider v1beta1 package)